packages/amplify-graphql-schema-generator/src/ts-schema-generator/helpers.ts

```diff
@@ -27,32 +27,46 @@ const GQL_TYPESCRIPT_DATA_SCHEMA_TYPE_MAP = {
  * @returns Typescript data schema property in TS Node format
  */
 const createProperty = (field: Field): ts.Node => {
-  const typeExpression = createDataType(field.type);
+  const typeExpression = createDataType(field);
   return ts.factory.createPropertyAssignment(ts.factory.createIdentifier(field.name), typeExpression as ts.Expression);
 };
 
 /**
  * Creates a typescript data schema type from internal SQL schema representation
  * Example typescript data schema type output: `a.string().required()`
- * @param type SQL IR field type
+ * @param field SQL IR field
  * @returns Typescript data schema type in TS Node format
  */
-const createDataType = (type: FieldType): ts.Node => {
-  if (type.kind === 'Scalar') {
+const createDataType = (field: Field): ts.Node => {
+  const sequenceRegex = /^nextval\(.+::regclass\)$/;
+  if (field.default?.kind === 'DB_GENERATED' && sequenceRegex.test(field.default.value.toString())) {
```

Comment:
We can possible move the check `sequenceRegex.test(field.default.value.toString()` to a separate method so that it can reused in the future like `isSequenceField(field)`.